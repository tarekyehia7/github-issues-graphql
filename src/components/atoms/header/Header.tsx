import React, { PropsWithChildren } from 'react';

import { HeaderStyled } from './Header.styled';

export const Header = ({ children }: PropsWithChildren) => {
	return <HeaderStyled>{children}</HeaderStyled>;
};
