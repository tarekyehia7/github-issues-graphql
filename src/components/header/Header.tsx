import React from 'react';
import styled from 'styled-components';

import { GithubIcon } from '../../icons/GithubIcon';

const GithubIco = styled(GithubIcon)`
    height: 2.5rem;
    width: 2.5rem;
    fill: #ffffff;
    margin: 1rem;
`;

const StyledHeader = styled.header`
    display: flex;
    justify-content: space-between;
    align-items: flex-start;
    flex-direction: column;
    background-color: #24292f;
`;

export const Header = () => {
    return (
        <StyledHeader>
            <GithubIco />
        </StyledHeader>
    );
};