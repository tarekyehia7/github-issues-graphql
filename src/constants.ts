export const constants = {
	repository: 'facebook',
	projectName: 'react',
	searchTerm: 'title,body',
	issuesPerPage: 11,
	githubUrl: 'https://github.com',
};
