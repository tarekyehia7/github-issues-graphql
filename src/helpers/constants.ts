export const constants = {
    repository: 'facebook',
    projectName: 'react',
	searchTerm: 'title,body',
    issuesPerPage: 10,
    githubUrl: 'https://github.com'
};