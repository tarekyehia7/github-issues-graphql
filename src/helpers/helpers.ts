import formatDistanceToNowStrict from 'date-fns/formatDistanceToNowStrict';

export const getPagesNumber = (totalIssuesCount: number, issuesPerPage: number): number => {
	const totalPages = Math.ceil(totalIssuesCount / issuesPerPage);
	return !totalPages ? 1 : totalPages;
};

export const formatDate = (date: string) => {
	try {
		const myDate = new Date(date);
		const myFormattedDate = formatDistanceToNowStrict(myDate, { addSuffix: true });
		return myFormattedDate;
	} catch (e) {
		console.error(e);
		return '';
	}
};
