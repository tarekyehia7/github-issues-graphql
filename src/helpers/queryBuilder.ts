import { StatusEnum } from "../pages/Issues/Issues";

interface Query {
    repo: string;
    input: string;
    is: string;
    in: string;
    status: StatusEnum;
}

export const buildQuery = (query: Query) => {
    return Object.entries(query)
        .reduce((str, [p, val]) => `${str}${p}:${val} `, '')
        .replace('status:', '')
        .replace('input:', '')
        .trim();
};