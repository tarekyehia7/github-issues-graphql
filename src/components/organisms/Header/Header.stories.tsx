import React from 'react';
import { withRouter } from 'storybook-addon-react-router-v6';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { Header } from './Header';

export default {
	title: 'organisms/Header',
	component: Header,
	decorators: [withRouter],
} as ComponentMeta<typeof Header>;

export const Primary: ComponentStory<typeof Header> = () => <Header />;
