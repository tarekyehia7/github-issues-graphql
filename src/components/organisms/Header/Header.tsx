import React from 'react';

import { HeaderStyled } from '../../atoms/header/Header.styled';
import { PagesLinks } from '../../molecules/PagesLinks/PagesLinks';
import { ReposHeaderLinks } from '../../molecules/RepostHeaderLinks/ReposHeaderLinks';
import { GithubIco, LinkContainer } from './Header.styled';

export const Header = () => {
	return (
		<>
			<HeaderStyled>
				<GithubIco />
			</HeaderStyled>
			<LinkContainer>
				<ReposHeaderLinks />
				<PagesLinks />
			</LinkContainer>
		</>
	);
};
