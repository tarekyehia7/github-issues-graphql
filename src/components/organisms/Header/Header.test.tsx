import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import 'jest-styled-components';

import { Header } from './Header';
import { PageWithTheme } from '../../../helpers/testing/helpers';
import { constants } from '../../../constants';

const HeaderWithTheme = () => {
	return (
		<PageWithTheme>
			<Header />
		</PageWithTheme>
	);
};

describe('<Header />', () => {
	it('Should match snapshot', () => {
		const { container } = render(<HeaderWithTheme />);
		expect(container).toMatchSnapshot();
	});

	it('Should have correct links', () => {
		const { githubUrl, repository, projectName } = constants;
		const { getAllByRole } = render(<HeaderWithTheme />);
		const links = getAllByRole('link');

		expect(links).toHaveLength(3);
		expect(links[0]).toHaveAttribute('href', `${githubUrl}/${repository}`);
		expect(links[1]).toHaveAttribute('href', `${githubUrl}/${repository}/${projectName}`);
		expect(links[2]).toHaveAttribute('href', `/`);
	});
});
