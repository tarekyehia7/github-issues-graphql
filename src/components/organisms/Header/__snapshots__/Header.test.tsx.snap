// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #24292f;
  min-height: 4rem;
}

.c5 {
  color: #0969da;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #0969da;
}

.c8 {
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 600;
  color: #24292f;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0 8px;
  font-size: 14px;
  line-height: 30px;
  text-align: center;
  white-space: nowrap;
  cursor: pointer;
  background-color: transparent;
  border: 0;
  border-radius: 6px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c8:after {
  position: absolute;
  right: 50%;
  bottom: calc(50% - 25px);
  width: 4rem;
  height: 2px;
  content: '';
  background: #fd8c73;
  border-radius: 6px;
  -webkit-transform: translate(50%,-50%);
  -ms-transform: translate(50%,-50%);
  transform: translate(50%,-50%);
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-bottom: 1rem;
}

.c4 {
  height: 1rem;
  width: 1rem;
  margin-top: 0.1rem;
}

.c6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 0 7px;
  font-size: 12px;
  font-weight: 500;
  line-height: 18px;
  white-space: nowrap;
  border: 1px solid transparent;
  border-radius: 2em;
  color: #57606a;
  border-color: #d0d7de;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 4rem;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding: 1rem;
  background-color: #f6f8fa;
}

.c1 {
  height: 2.5rem;
  width: 2.5rem;
  fill: #ffffff;
  margin: 1rem;
}

<div>
  <header
    class="c0"
  >
    <svg
      class="c1"
      preserveAspectRatio="xMidYMid meet"
      role="img"
      version="1.0"
      viewBox="0 0 120.000000 120.000000"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        transform="translate(0.000000,120.000000) scale(0.100000,-0.100000)"
      >
        <path
          d="M452 1170 c-109 -29 -187 -74 -272 -160 -87 -86 -131 -163 -160 -276 -36 -141 -18 -294 52 -427 48 -92 164 -205 259 -251 103 -51 113 -48 117 35 l3 66 -59 0 c-75 0 -116 23 -147 84 -13 26 -38 58 -55 73 -36 29 -31 51 9 41 32 -8 51 -23 47 -38 -1 -7 -1 -8 2 -4 5 9 32 -11 32 -25 0 -5 -4 -6 -10 -3 -5 3 -10 1 -10 -5 0 -7 6 -10 14 -7 8 3 22 -3 31 -13 39 -43 137 -37 153 8 6 15 13 34 17 43 5 13 -2 18 -35 23 -60 9 -135 50 -165 90 -60 79 -73 213 -29 306 19 39 26 69 25 103 -2 26 0 61 4 77 9 40 43 40 119 1 55 -28 59 -28 132 -18 53 8 95 8 148 0 73 -10 77 -10 132 18 76 39 110 39 119 -1 4 -16 6 -51 4 -77 -1 -34 6 -64 25 -103 44 -93 31 -227 -29 -306 -31 -41 -108 -82 -169 -90 l-45 -7 17 -25 c13 -20 18 -57 22 -151 6 -151 6 -151 119 -95 95 46 211 159 259 251 42 80 72 198 72 283 0 85 -30 203 -72 283 -42 81 -164 203 -245 245 -132 69 -291 88 -431 52z"
        />
        <path
          d="M290 240 c0 -5 5 -10 10 -10 6 0 10 5 10 10 0 6 -4 10 -10 10 -5 0 -10 -4 -10 -10z"
        />
        <path
          d="M330 220 c0 -5 5 -10 10 -10 6 0 10 5 10 10 0 6 -4 10 -10 10 -5 0 -10 -4 -10 -10z"
        />
        <path
          d="M425 220 c-3 -5 1 -10 10 -10 9 0 13 5 10 10 -3 6 -8 10 -10 10 -2 0 -7 -4 -10 -10z"
        />
        <path
          d="M380 210 c0 -5 5 -10 10 -10 6 0 10 5 10 10 0 6 -4 10 -10 10 -5 0 -10 -4 -10 -10z"
        />
      </g>
    </svg>
  </header>
  <div
    class="c2"
  >
    <div
      class="c3"
    >
      <svg
        aria-hidden="true"
        class="c4"
        data-view-component="true"
        version="1.1"
        viewBox="0 0 16 16"
      >
        <path
          d="M2 2.5A2.5 2.5 0 014.5 0h8.75a.75.75 0 01.75.75v12.5a.75.75 0 01-.75.75h-2.5a.75.75 0 110-1.5h1.75v-2h-8a1 1 0 00-.714 1.7.75.75 0 01-1.072 1.05A2.495 2.495 0 012 11.5v-9zm10.5-1V9h-8c-.356 0-.694.074-1 .208V2.5a1 1 0 011-1h8zM5 12.25v3.25a.25.25 0 00.4.2l1.45-1.087a.25.25 0 01.3 0L8.6 15.7a.25.25 0 00.4-.2v-3.25a.25.25 0 00-.25-.25h-3.5a.25.25 0 00-.25.25z"
        />
      </svg>
      <strong>
         
        <a
          class="c5"
          href="https://github.com/facebook"
        >
           
          facebook
           
        </a>
         
      </strong>
      /
      <strong>
         
        <a
          class="c5"
          href="https://github.com/facebook/react"
        >
          react
        </a>
         
      </strong>
      <label
        class="c6"
      >
        Public
      </label>
    </div>
    <div
      class="c7"
    >
      <a
        class="c8"
        href="/"
      >
        Issues
      </a>
    </div>
  </div>
</div>
`;
